crates/meroctl/src/cli/config.rs

```diff
@@ -0,0 +1,188 @@
+
+                for protocol in addr.iter() {
+                    match (&protocol, ipv4_host, ipv6_host, self.swarm_port) {
+                        (Protocol::Ip4(_), Some(ipv4_host), _, _) => {
+                            new_addr.push(Protocol::Ip4(ipv4_host));
+                        }
+                        (Protocol::Ip6(_), _, Some(ipv6_host), _) => {
+                            new_addr.push(Protocol::Ip6(ipv6_host));
+                        }
+                        (Protocol::Tcp(_) | Protocol::Udp(_), _, _, Some(new_port)) => {
+                            new_addr.push(match &protocol {
+                                Protocol::Tcp(_) => Protocol::Tcp(new_port),
+                                Protocol::Udp(_) => Protocol::Udp(new_port),
+                                _ => unreachable!(),
+                            });
+                        }
+                        _ => new_addr.push(protocol),
+                    }
+                }
+
+                *item = Value::from(new_addr.to_string());
+            }
+        }
+
+        // Update server listen addresses
+        if !self.server_host.is_empty() || self.server_port.is_some() {
+            let server_table = doc["server"].as_table_mut().unwrap();
+            let listen_array = server_table["listen"].as_array_mut().unwrap();
+
+            for item in listen_array.iter_mut() {
+                let addr: Multiaddr = item.as_str().unwrap().parse().unwrap();
+                let mut new_addr = Multiaddr::empty();
+
+                for protocol in addr.iter() {
+                    match (&protocol, ipv4_host, ipv6_host, self.server_port) {
+                        (Protocol::Ip4(_), Some(ipv4_host), _, _) => {
+                            new_addr.push(Protocol::Ip4(ipv4_host));
+                        }
+                        (Protocol::Ip6(_), _, Some(ipv6_host), _) => {
+                            new_addr.push(Protocol::Ip6(ipv6_host));
+                        }
+                        (Protocol::Tcp(_), _, _, Some(new_port)) => {
+                            new_addr.push(Protocol::Tcp(new_port));
+                        }
+                        _ => new_addr.push(protocol),
+                    }
+                }
+
+                *item = Value::from(new_addr.to_string());
+            }
+        }
+
+        // Update boot nodes if provided
+        if !self.boot_nodes.is_empty() {
+            let bootstrap_table = doc["bootstrap"].as_table_mut().unwrap();
+            let list_array = bootstrap_table["nodes"].as_array_mut().unwrap();
+            list_array.clear();
+            for node in self.boot_nodes.iter() {
+                list_array.push(node.to_string());
+            }
+        } else if let Some(network) = self.boot_network {
+            let bootstrap_table = doc["bootstrap"].as_table_mut().unwrap();
+            let list_array = bootstrap_table["nodes"].as_array_mut().unwrap();
```

Comment:
also, I don't think you need to break these into separate statements with individual `unwrap`s

as highlighted here - https://github.com/calimero-network/core/pull/405#discussion_r1658368702

`doc["bootstrap"]["nodes"]` accesses the same item